src/main/kotlin/eu/kevin/api/Serializer.kt

```diff
@@ -0,0 +1,13 @@
+package eu.kevin.api
+
+import com.squareup.moshi.Moshi
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+
+internal class Serializer(private val moshi: Moshi) {
+    inline fun <reified T> serialize(obj: T): String = moshi.adapter(T::class.java).toJson(obj)
+
+    suspend inline fun <reified T> deserialize(response: String): T = withContext(Dispatchers.IO) {
```

Comment:
will the context ever be outside IO thread pool? because this will be called from Ktor anyway and it will just use the active thread pool (which will likely be IO). I would still be using kotlinx.serialization

Reply:
It will not be called from ktor since deserialization happens after the response has been received. Probably doesn't matter still